src/v2/controllers/FormController.js

```diff
@@ -153,7 +153,9 @@ export default Controller.extend({
       // TODO: OKTA-243167 what's the approach to show spinner indicating API in flight?
       actionFn()
         .then((resp) => {
-          if (actionPath === 'cancel' && this.options.settings.get('useInteractionCodeFlow')) {
+          if (actionPath === 'cancel' && window.location.pathname === '/signin/forgot-password') {
```

Comment:
Maybe this should just be based on `useInteractionCodeFlow` ? Then `Back to Sign in` will go to the origin even if this flow is running on a different URL (like authorize endpoint).

```
const { settings, appState } = options;
if (actionPath === 'cancel') }
  if (settings.get('useInteractionCodeFlow')) {
      // embedded
      appState.trigger('restartLoginFlow');
  } else {
     // federated
      window.location.assign(window.location.origin);
  }
}

```

Reply:
@aarongranick-okta then every 'cancel' will be a page reload

Reply:
Isn't that what we need to get a new stateToken